src/webgpu/shader/execution/expression/binary/f32_logical.spec.ts

```diff
@@ -0,0 +1,246 @@
+  .uniqueId('xxxxxxxxx')
+  .specURL('https://www.w3.org/TR/WGSL/#floating-point-evaluation')
+  .desc(
+    `
+Expression: x == y
+Accuracy: Correct result
+`
+  )
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    const truthFunc = (lhs: Scalar, rhs: Scalar): boolean => {
+      return (lhs.value as number) === (rhs.value as number);
+    };
+
+    let cases: Array<Case> = [];
+    const numeric_range = fullNumericRange();
+    numeric_range.forEach(lhs => {
+      numeric_range.forEach(rhs => {
+        cases = cases.concat(makeCase(lhs, rhs, truthFunc));
+      });
+    });
+
+    run(t, binary('=='), [TypeF32, TypeF32], TypeBool, t.params, cases);
+  });
+
+g.test('not_equals')
+  .uniqueId('xxxxxxxxx')
+  .specURL('https://www.w3.org/TR/WGSL/#floating-point-evaluation')
+  .desc(
+    `
+Expression: x != y
+Accuracy: Correct result
+`
+  )
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    const truthFunc = (lhs: Scalar, rhs: Scalar): boolean => {
+      return (lhs.value as number) !== (rhs.value as number);
+    };
+
+    let cases: Array<Case> = [];
+    const numeric_range = fullNumericRange();
+    numeric_range.forEach(lhs => {
+      numeric_range.forEach(rhs => {
+        cases = cases.concat(makeCase(lhs, rhs, truthFunc));
+      });
+    });
+
+    run(t, binary('!='), [TypeF32, TypeF32], TypeBool, t.params, cases);
+  });
+
+g.test('less_than')
+  .uniqueId('xxxxxxxxx')
+  .specURL('https://www.w3.org/TR/WGSL/#floating-point-evaluation')
+  .desc(
+    `
+Expression: x < y
+Accuracy: Correct result
+`
+  )
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    const truthFunc = (lhs: Scalar, rhs: Scalar): boolean => {
+      return (lhs.value as number) < (rhs.value as number);
+    };
+
+    let cases: Array<Case> = [];
+    const numeric_range = fullNumericRange();
+    numeric_range.forEach(lhs => {
+      numeric_range.forEach(rhs => {
+        cases = cases.concat(makeCase(lhs, rhs, truthFunc));
+      });
+    });
+
+    run(t, binary('<'), [TypeF32, TypeF32], TypeBool, t.params, cases);
+  });
+
+g.test('less_equals')
+  .uniqueId('xxxxxxxxx')
+  .specURL('https://www.w3.org/TR/WGSL/#floating-point-evaluation')
+  .desc(
+    `
+Expression: x <= y
+Accuracy: Correct result
+`
+  )
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    const truthFunc = (lhs: Scalar, rhs: Scalar): boolean => {
+      return (
+        (lhs.value as number) < (rhs.value as number) ||
```

Comment:
Out of curiosity, why not `<=`? You're using a non-strict comparison here anyway.

Reply:
Originally wrote this doing bit operations, so this was just legacy from that. I agree `<=` makes more sense